pkg/pool/pool.go

```diff
@@ -40,6 +46,12 @@ type BucketedBytes struct {
 	new func(s int) *[]byte
 }
 
+func (p *BucketedBytes) UsedBytes() uint64 {
+	p.mtx.Lock()
```

Comment:
Maybe we can use a RWLock?

Reply:
I did, sadly we are also theoretically writing in the other two methods, so we can use the RLock only here.